spec/lang/representation.md

```diff
@@ -146,14 +131,16 @@ impl Type {
             throw!();
         }
         let remaining_bytes = bytes.subslice_with_length(M::T::PTR_SIZE.bytes(), bytes.len() - M::T::PTR_SIZE.bytes());
-        let meta = decode_ptr_meta::<M>(ptr_type.meta_kind(), remaining_bytes)?;
-        ret(Value::Ptr(ptr.widen(meta)))
+        if !remaining_bytes.is_empty() {
+            throw!();
+        }
+        ret(Value::Ptr(ptr.widen(None)))
```

Comment:
Can you revert this to how it was before the PR? The entire subslicing thing is unnecessary. (And I think when metadata becomes a thing, we can find a way to write this that uses less subslicing -- ideally by sharing some logic with the encode/code for tuples.)